underlay/js/dataTables.buttons.min.js

```diff
@@ -1,39 +1,44 @@
+w&&(g=l-(g+w),d-=x+w,w=g>d?w+d:w+g)):(w=l-g,d<x+w&&(g=l-(g+w),d-=x+w,w=g>d?w+d:w+g));c.css("left",c.position().left+w)}else"absolute"===g?(l=h.position(),c.css({top:l.top+h.outerHeight(),left:l.left}),k=c.outerHeight(),g=h.offset().top,w=0,w=h.offset().left,B=h.outerWidth(),B=w+B,g=c.offset().left,x=a.width(),x=g+x,v=l.top-k-5,n=d.offset().top+d.height(),p=l.top+h.outerHeight()+k,n=p-n,p=l.top-k,r=d.offset().top,(n>r-p||e.dropup)&&-v<r&&c.css("top",v),w="button-right"===e.align?B-x:w-g,c.css("left",
+c.position().left+w)):(g=c.height()/2,g>f(A).height()/2&&(g=f(A).height()/2),c.css("marginTop",-1*g));e.background&&u.background(!0,e.backgroundClassName,e.fade,h);f("div.dt-button-background").on("click.dtb-collection",function(){});f("body").on("click.dtb-collection",function(C){var I=f.fn.addBack?"addBack":"andSelf",J=f(C.target).parent()[0];(!f(C.target).parents()[I]().filter(a).length&&!f(J).hasClass("dt-buttons")||f(C.target).hasClass("dt-button-background"))&&m()}).on("keyup.dtb-collection",
+function(C){27===C.keyCode&&m()});e.autoClose&&setTimeout(function(){b.on("buttons-action.b-internal",function(C,I,J,O){O[0]!==h[0]&&m()})},0);f(c).trigger("buttons-popover.dt")}});u.background=function(a,b,c,d){c===t&&(c=400);d||(d=y.body);a?E(f("<div/>").addClass(b).css("display","none").insertAfter(d),c):F(f("div."+b),c,function(){f(this).removeClass(b).remove()})};u.instanceSelector=function(a,b){if(a===t||null===a)return f.map(b,function(h){return h.inst});var c=[],d=f.map(b,function(h){return h.name}),
+e=function(h){if(Array.isArray(h))for(var m=0,g=h.length;m<g;m++)e(h[m]);else"string"===typeof h?-1!==h.indexOf(",")?e(h.split(",")):(h=f.inArray(h.trim(),d),-1!==h&&c.push(b[h].inst)):"number"===typeof h&&c.push(b[h].inst)};e(a);return c};u.buttonSelector=function(a,b){for(var c=[],d=function(g,l,k){for(var n,p,r=0,v=l.length;r<v;r++)if(n=l[r])p=k!==t?k+r:r+"",g.push({node:n.node,name:n.conf.name,idx:p}),n.buttons&&d(g,n.buttons,p+"-")},e=function(g,l){var k,n=[];d(n,l.s.buttons);var p=f.map(n,function(r){return r.node});
+if(Array.isArray(g)||g instanceof f)for(p=0,k=g.length;p<k;p++)e(g[p],l);else if(null===g||g===t||"*"===g)for(p=0,k=n.length;p<k;p++)c.push({inst:l,node:n[p].node});else if("number"===typeof g)c.push({inst:l,node:l.s.buttons[g].node});else if("string"===typeof g)if(-1!==g.indexOf(","))for(n=g.split(","),p=0,k=n.length;p<k;p++)e(n[p].trim(),l);else if(g.match(/^\d+(\-\d+)*$/))p=f.map(n,function(r){return r.idx}),c.push({inst:l,node:n[f.inArray(g,p)].node});else if(-1!==g.indexOf(":name"))for(g=g.replace(":name",
```

Comment:
The variables on this particular line feels a bit too short. A bit more verbosity needed.

Reply:
;)